Add Southern company Integration (#26179)
home-assistant/home-assistant.github.io: source/_integrations/southern_company.markdown

---
title: Southern Company
description: Instructions on how to integrate Southern Company within Home Assistant.
ha_category:
  - Energy
  - Sensor
ha_release: 2023.3
ha_iot_class: Cloud Polling
ha_codeowners:
  - '@Lash-L'
ha_domain: southern_company
ha_config_flow: true
ha_platforms:
  - sensor
ha_integration_type: integration
---

> The Southern Company sensor platform allows you to get energy information from [Southern Company Energy Companies.](https://www.southerncompany.com/about/our-companies.html)

Reviewer: ```suggestion
The Southern Company sensor platform allows you to get energy information from [Southern Company Energy Companies](https://www.southerncompany.com/about/our-companies.html).
```

> The Southern Company sensor platform allows you to get energy information from [Southern Company Energy Companies](https://www.southerncompany.com/about/our-companies.html).

Reviewer 2: ```suggestion
The Southern Company integration allows you to get energy information from [Southern Company Energy Companies](https://www.southerncompany.com/about/our-companies.html).
```

Supported companies (tested):

- [Georgia Power](https://www.georgiapower.com/)

Untested companies:

- [Alabama Power](https://www.alabamapower.com/)
- [Mississippi Power](https://www.mississippipower.com/)

Southern Company natural gas companies are not currently supported.

> To add this integration to your installation, You will need your Southern Company login email and password.

Reviewer 3: s/You/you/

Reviewer 3: And is it email or username?

> To add this platform to your installation, You will need your southern company login email and password.

Reviewer: ```suggestion
To add this platform to your installation, You will need your Southern Company login email and password.
```

> To add this platform to your installation, You will need your Southern Company login email and password.

Reviewer 2: ```suggestion
To add this integration to your installation, You will need your Southern Company login email and password.
```

{% include integrations/config_flow.md %}

## Sensors

- Monthly cost: The most recent update of your current bill total
- Monthly consumption: The most recent update of your current energy usage
- Average daily usage: How much energy you have used on average in this billing cycle
- Average daily cost: How much it cost on average to run your electricity in this billing cycle

> - Lower Projected monthly Usage: The lower bounds of how much energy southern company projects you will use.

Reviewer: ```suggestion
- Lower Projected monthly Usage: The lower bounds of how much energy Southern Company projects you will use
```

> - Higher projected monthly usage: The upper bounds of how much energy southern company projects you will use.

Reviewer: ```suggestion
- Higher projected monthly usage: The upper bounds of how much energy Southern Company projects you will use
```

> - Lower projected monthly cost: The lower bounds of how much energy southern company projects your bill will cost

Reviewer: ```suggestion
- Lower projected monthly cost: The lower bounds of how much energy Southern Company projects your bill will cost
```

> - Monthly cost: The most recent update of your current bill total
> - Monthly consumption: The most recent update of your current energy usage
> - Average daily usage: How much energy you have used on average in this billing cycle
> - Average daily cost: How much it cost on average to run your electricity in this billing cycle
> - Lower Projected monthly Usage: The lower bounds of how much energy Southern Company projects you will use
> - Higher projected monthly usage: The upper bounds of how much energy Southern Company projects you will use
> - Lower projected monthly cost: The lower bounds of how much energy Southern Company projects your bill will cost
> - Higher projected monthly cost: The upper bounds of how much energy southern company projects your bill will cost

Reviewer 2: ```suggestion
- Monthly cost: The most recent update of your current bill total.
- Monthly consumption: The most recent update of your current energy usage.
- Average daily usage: How much energy you have used on average in this billing cycle.
- Average daily cost: How much it costs on average to run your electricity in this billing cycle.
- Lower Projected monthly Usage: The lower bounds of how much energy Southern Company projects you will use.
- Higher projected monthly usage: The upper bounds of how much energy Southern Company projects you will use.
- Lower projected monthly cost: The lower bounds of how much energy Southern Company projects your bill will cost.
- Higher projected monthly cost: The upper bounds of how much energy southern company projects your bill will cost.
```

## Energy

> Because Southern Company only releases usage/cost data with a 48 hour delay, data is inserted into a statistic object. To add the statistic object to your Energy dashboard, you need to select 'southern_company:energy_cost_{ACCOUNT_NUMBER}' for the energy cost and 'southern_company:energy_usage_{ACCOUNT_NUMBER}' for the energy usage.

Reviewer 3: Can you add delayed data in the energy dashboard? https://community.home-assistant.io/t/import-historic-sensor-data/340980 suggests you cannot although things might have changed since Sep '21.

Author: Yes. I copy what tibber, an already implemented integration does

Reviewer 2: > Can you add delayed data in the energy dashboard?

This is not related to the contents of this PR.

Reviewer 3: Sorry my question wasn't a feature request. It was a clarification question if what @Lash-L is doing actually works. So the energy dashboard keeps getting updated and shows a gap for the past 2 days? Or is the data time shifted by 2 days? I'm asking because my energy provider also provides data delayed by 2 days and https://community.home-assistant.io/t/pg-e-energy-monitoring-integration/336038/38 suggested it wouldn't be possible to import that data in the dashboard.

Author: @tronikos message me on the home assistant community forum. My username is the same as here Lash-L. This isn't the place to talk about this as every time a response is made fabaff and Frenck are getting notified
